### ISSUE TITLE
[Accessibility][Internationalization] Talkback doesn't work properly when text is localized to Chinese on Samsung devices when using the Samsung text-to-speech engine

### ISSUE THREAD

**Issue by danagbemava-nc** [flutter/flutter#179491]
### Steps to reproduce

1. Get and run the latest version of the wonderous app
2. Enable `Talkback`
3. Navigate through the app with English as the language
4. Observe that the text is read properly
5. Go to the settings and change the language to Chinese
6. Navigate through the app

### Expected results

The localized Chinese text is read properly

### Actual results

The text is read in English and in some cases, buttons are read out instead of text. 

This seems to be due to the "Samsung text-to-speech" engine. When I tested with the "Speech Recognition and Synthesis from Google" engine, the output was better. It read in Chinese but it still read the component name afterwards (not like on my pixel or iOS)

I downloaded the TTS languages for Chinese as well, but it didn't seem to have any effect when I was using the Samsung TTS engine. 

I tested this on a Samsung A03 (android 12), S10+(android 12) and S23(android 16) and the results were the same. 

### Code sample

The wonderous app should suffice

### Screenshots or Video

| | | 
| -- | -- |
| <video src="https://github.com/user-attachments/assets/861c1695-d877-48a7-9344-01d563162740" /> | <video src="https://github.com/user-attachments/assets/d119c716-2397-48b8-a84b-3944770f6bdd" /> |

### Logs

N/A

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.38.4, on macOS 26.2 25C56 darwin-arm64, locale en-US) [756ms]
    • Flutter version 3.38.4 on channel stable at /Users/deanli/dev/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 66dd93f9a2 (33 hours ago), 2025-12-03 14:56:10 -0800
    • Engine revision a5cb96369e
    • Dart version 3.10.3
    • DevTools version 2.51.1
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, omit-legacy-version-file, enable-lldb-debugging, enable-uiscene-migration

[✓] Android toolchain - develop for Android devices (Android SDK version 36.1.0) [6.5s]
    • Android SDK at /Users/deanli/Library/Android/sdk
    • Emulator version 36.3.9.0 (build_id 14404213) (CL:N/A)
    • Platform android-36.1, build-tools 36.1.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.8+-14196175-b1038.72)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 26.2) [1,632ms]
    • Xcode at /Applications/Xcode-26.2.0-release-candidate.app/Contents/Developer
    • Build 17C48
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [5ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (6 available) [7.4s]
    • SM A035F (mobile)                     • R9ZT902JTGA               • android-arm64  • Android 12 (API 31)
    • iPhone 16 Pro Max (wireless) (mobile) • 00008140-00194CC23A68801C • ios            • iOS 26.2 23C52
    • Nexus (wireless) (mobile)             • 00008020-001875E83A38002E • ios            • iOS 18.7.2 22H124
    • Dean iPad (wireless) (mobile)         • 00008103-000825C811E3401E • ios            • iOS 26.2 23C52
    • macOS (desktop)                       • macos                     • darwin-arm64   • macOS 26.2 25C56 darwin-arm64
    • Chrome (web)                          • chrome                    • web-javascript • Google Chrome 143.0.7499.40

[✓] Network resources [763ms]
    • All expected network resources are available.

• No issues found!
```

```
[!] Flutter (Channel master, 3.39.0-1.0.pre-426, on macOS 26.2 25C56 darwin-arm64, locale en-US) [2.1s]
    • Flutter version 3.39.0-1.0.pre-426 on channel master at /Users/deanli/dev/master
    ! Warning: `flutter` on your path resolves to /Users/deanli/dev/stable/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/deanli/dev/master. Consider adding /Users/deanli/dev/master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/deanli/dev/stable/bin/dart, which is not inside your current Flutter SDK checkout at /Users/deanli/dev/master. Consider adding /Users/deanli/dev/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d50dfb9bf7 (6 hours ago), 2025-12-04 20:08:52 -0500
    • Engine revision d50dfb9bf7
    • Dart version 3.11.0 (build 3.11.0-208.0.dev)
    • DevTools version 2.53.0
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, omit-legacy-version-file, enable-windowing, enable-lldb-debugging, enable-uiscene-migration
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 36.1.0) [1,515ms]
    • Android SDK at /Users/deanli/Library/Android/sdk
    • Emulator version 36.3.9.0 (build_id 14404213) (CL:N/A)
    • Platform android-36.1, build-tools 36.1.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.8+-14196175-b1038.72)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 26.2) [1,096ms]
    • Xcode at /Applications/Xcode-26.2.0-release-candidate.app/Contents/Developer
    • Build 17C48
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [5ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (6 available) [6.4s]
    • SM A035F (mobile)                     • R9ZT902JTGA               • android-arm64  • Android 12 (API 31)
    • iPhone 16 Pro Max (wireless) (mobile) • 00008140-00194CC23A68801C • ios            • iOS 26.2 23C52
    • Nexus (wireless) (mobile)             • 00008020-001875E83A38002E • ios            • iOS 18.7.2 22H124
    • Dean iPad (wireless) (mobile)         • 00008103-000825C811E3401E • ios            • iOS 26.2 23C52
    • macOS (desktop)                       • macos                     • darwin-arm64   • macOS 26.2 25C56 darwin-arm64
    • Chrome (web)                          • chrome                    • web-javascript • Google Chrome 143.0.7499.40

[✓] Network resources [846ms]
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>
